File: python/5kyu/rgb_to_hex_conversion.py
# https://www.codewars.com/kata/513e08acc600c94f01000001

def rgb(r, g, b):
    return '%02X%02X%02X' % tuple(max(0, min(x, 255)) for x in (r, g, b))


print(rgb(1, 2, 3), "010203")
print(rgb(-20, 275, 125), "00FF7D")

File: go/5kyu/rgb_to_hex_conversion.go
// https://www.codewars.com/kata/513e08acc600c94f01000001

package main

import (
	"fmt"
)

func RGB(r, g, b int) string {
	r, g, b = clamp(r), clamp(g), clamp(b)
	rgbHex := fmt.Sprintf("%02X%02X%02X", r, g, b)
	return rgbHex
}

func clamp(value int) int {
	if value < 0 {
		return 0
	}
	if value > 255 {
		return 255
	}
	return value
}

func main() {
	fmt.Println(RGB(254, 253, 252), "FEFDFC")
	fmt.Println(RGB(-20, 275, 125), "00FF7D")
}
